src/Storages/MergeTree/MergeTreeReaderCompact.cpp

```diff
@@ -239,4 +240,53 @@ bool MergeTreeReaderCompact::isContinuousReading(size_t mark, size_t column_posi
         || (mark == last_mark + 1 && column_position == 0 && last_column == data_part->getColumns().size() - 1);
 }
 
+
+size_t MergeTreeReaderCompact::getReadBufferSize()
+{
+    size_t buffer_size = 0;
+    size_t columns_num = columns.size();
+    size_t last_column_position = data_part->getColumns().size() - 1;
+    size_t file_size = data_part->getFileSizeOrZero(MergeTreeDataPartCompact::DATA_FILE_NAME_WITH_EXTENSION);
+    size_t marks_count = data_part->getMarksCount();
+
+    auto next = [&](size_t & row_index, size_t & column_index)
```

Comment:
It would be really nice to implement iterator here. Cause this lambda captures all variables, and some of them changed later in unpredictable way. Readability of it is pretty low.